### DISCUSSION TITLE
Avoid usage of Rec.Count() if possible

### DISCUSSION THREAD

**Discussion by christophstuber** [StefanMaron/BusinessCentral.LinterCop#722] [New Rule]
Count can be one of the most expensive database operations on large tables. I often see examples like this in code:

### Checking for no records:

```AL
if Rec.Count() = 0 then
```

Should be replaced by:

```AL
if Rec.IsEmpty() then
```

### Checking for any records:

```AL
if Rec.Count() > 0 then
```

Becomes:

```AL
if not Rec.IsEmpty() then
```

### Checking for exactly one record:

```AL
if Rec.Count() = 1 then
```

Refering to https://vjeko.com/2017/03/08/the-if-count-1-conundrum/

```AL
if Rec.Find('-') and (Rec.Next() = 0) then
```

### Checking for more than one record:

```AL
if Rec.Count() > 1 then
```

```AL
if Rec.Find('-') and (Rec.Next() <> 0) then
```

**Comment by Arthurvdv** [comment]
@christophstuber, this idea was long overdue where I've today decided to create the rule for this.

I've ended up splitsing this into two separate rules. The reason for this, is that I believe the first one (LC0081) could be set as a warning for all projects, where the second one (LC0082) is a balance between performance and readability of the code. In this way it's more flexible on which of these rules want to be enforced.

* [LC0081 - Use Rec.IsEmpty() for checking record existence.](https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0081)
* [LC0082 - Use Rec.Find('-') with Rec.Next() for checking exactly one record.](https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0082)

Which both is now available in the [v0.31.4](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.31.4) (pre)release version of the LinterCop.

Thank you for creating this idea, especially the part on the Rec.Find('-') with Rec.Next(), that is a nice catch.

**Comment by Arthurvdv** [comment]
The rules are now available from the release version [v0.32.0](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.32.0) of the LinterCop.
